src/docs/develop/variables.md

```diff
@@ -2,29 +2,41 @@
 title: Variables
 ---
 
-Service variables are provided whenever you build, deploy, or run `railway run`. When
-defined, they are made available to your application at runtime as environment variables.
+Variables provide a powerful way to manage configuration and secrets across services in Railway.
 
-Variables are made available in the following scenarios:
+When defined, they are made available to your application as environment variables in the following scenarios:
 
 - The build process for each service deployment.
 - The running service deployment.
 - The command invoked by `railway run <COMMAND>`
 - The local shell via `railway shell`
 
+In Railway, there is also a notion of configuration variables which allow you to control the behavior of the platform.
+
+## Types of Variables
+
+There are four main types of variables in Railway - 
+- **Service-level variables** are configured in your services' settings.
+- **Shared variables** are configured in your projects' settings and can be made available to any service in specific project.
+- **Railway-provided variables** are set by Railway to provide metadata about the environment, project, or service.
+- **User-provided configuration variables** are a group of reserved variables that change the behavior of Railway when set.
+
```

Comment:
I feel a bit overwhelmed by this section being upfront

Reply:
agreed, axed